app/lib/user_settings_decorator.rb

```diff
@@ -8,7 +8,7 @@ def initialize(user)
   end
 
   def update(settings)
-    @settings = settings
+    @settings = decoerce_values(settings)
```

Comment:
Instead of adding decoercing operation, let's use `ActiveModel::Type::Boolean.new.cast` in `coerce_values`. It accepts both of boolean and numbers.